client/src/modules/libraries/index.jsx

```diff
@@ -28,18 +28,19 @@ class Libraries extends PureComponent {
     }
   }
 
-  getLibraries = () => {
+  getLibraries = async () => {
     const { fetchLibraries } = this.props;
 
-    this.setState({ pending: true });
-    this.pendingPromise = makeAbortablePromise(fetchLibraries());
-    this.pendingPromise.promise
-      .then(() => this.setState({ pending: false }))
-      .catch(error => {
-        if (!(error instanceof AbortPromiseException)) {
-          this.setState({ pending: false });
-        }
-      });
+    try {
+      this.setState({ pending: true });
```

Comment:
this should go above try block

Reply:
it is waiting for the result: if you have await you don't proceed next lines of code until the awaited code is finished